### ISSUE TITLE
No changelog for 0.2.0 release

### ISSUE THREAD

**Issue by penguinpee** [has2k1/scikit-misc#23]
I have been notified of the latest release (0.2.0) on PyPI. However, there's no changelog entry. I believe the switch to Meson deserves some mention. I haven't tried building 0.2.0 yet. Time permitting, I will look into this over the weekend.

**Comment by penguinpee** [comment]
The [installation instructions](https://has2k1.github.io/scikit-misc/stable/installation.html) probably also need to be adopted.

**Comment by penguinpee** [comment]
Hmm. The new release tarball on PyPI uses and underscore instead of a dash (`scikit_misc` vs. `scikit-misc`). Is there any particular reason for that change?

Not sure if this should go in a separate issue. Let me know and I'll open another one.

**Comment by has2k1** [comment]
> The new release tarball on PyPI uses and underscore instead of a dash (scikit_misc vs. scikit-misc)

That is due to a change in how the package is built. Switched to https://github.com/pypa/build, which creates the source distribution first and then builds the wheel from it. I guess, since wheels must use underscores the source distribution is created with an underscore too. I am not sure if this agreement between the two is part of a PEP standard!

**Comment by has2k1** [comment]
> The [installation instructions](https://has2k1.github.io/scikit-misc/stable/installation.html) probably also need to be adopted.

Thank you. I missed those.

**Comment by has2k1** [comment]
> I have been notified of the latest release (0.2.0) on PyPI. However, there's no changelog entry. I believe the switch to Meson deserves some mention. I haven't tried building 0.2.0 yet. Time permitting, I will look into this over the weekend.

There is a changelog. The documentation is lagging a bit.

**Comment by penguinpee** [comment]
> I am not sure if this agreement between the two is part of a PEP standard!

Neither am I. Let me ask the Fedora Python list. It's not really an issue, more nuisance, since the naming guidelines require me to use dashes.

> There is a changelog. The documentation is lagging a bit.

I see. It's [there](https://has2k1.github.io/scikit-misc/stable/changelog.html) now. Thank you.

I switched the Fedora package to Meson over the weekend. Haven't pushed the update yet. It was a bit of work, but it looks fine now. One thing I noticed is the absence of tests. Is that planned for an upcoming release?

**Comment by penguinpee** [comment]
> I guess, since wheels must use underscores the source distribution is created with an underscore too.

Actually, where does it say that wheels need to use underscores? I just realized that what you are using is the standard Python build command. I've used that before and Fedora uses it for Python packages to create wheels and _not_ having underscores in the package names is not an issue.

Also in `pyproject.toml` `name` is set to [`scikit-misc`](https://github.com/has2k1/scikit-misc/blob/8f829888ecf0fa96eb18aeb3f723fc985de5abcd/pyproject.toml#L3) (with a dash).

I'm sure I'm missing something. Just wanna check before forwarding the question.

**Comment by has2k1** [comment]
> Actually, where does it say that wheels need to use underscores?

[PEP 427](https://peps.python.org/pep-0427/#escaping-and-unicode) dictates that the distribution name `scikit-misc` becomes `scikit_misc`.

**Comment by has2k1** [comment]
> One thing I noticed is the absence of tests.

The [tests](https://github.com/has2k1/scikit-misc/tree/main/skmisc/loess/tests) are in the sub-package.

**Comment by penguinpee** [comment]
> [PEP 427](https://peps.python.org/pep-0427/#escaping-and-unicode) dictates that the distribution name `scikit-misc` becomes `scikit_misc`

Thanks. I must have been confused by sub packages. So, underscore it is. Not a problem.

> The [tests](https://github.com/has2k1/scikit-misc/tree/main/skmisc/loess/tests) are in the sub-package

I saw that. But now that the build is driven by Meson, I should be able to run `meson test` for the tests to be executed. However, that comes back with:

```
ninja: no work to do.
No tests defined.
```

**Comment by eli-schwartz** [comment]
> I saw that. But now that the build is driven by Meson, I should be able to run `meson test` for the tests to be executed. However, that comes back with:

Not necessarily... the meson.build doesn't implement meson's own `test()` function. This project does use [spin](https://github.com/scientific-python/spin) though, which powers the `make test` target of the Makefile.

Under the hood it's just running `pytest skmisc` with the PYTHONPATH set up to a staging install.

**Comment by penguinpee** [comment]
> Under the hood it's just running `pytest skmisc` with the PYTHONPATH set up to a staging install.

Yeah. I saw that. In the last realease, which was still using Python tools, I had it set up in a special way due to RPM build environment. I guess I tripped over something else and then tried the direct approach using `meson test`. I have it fixed no and tests are being run once again. Thanks for being my (rubber) :duck: @eli-schwartz :wink:.

An updated Fedora package will be pushed shortly. Thank you all for the assistance and clarification. I'll close the issue.
